File: src/main/java/no/nav/fo/veilarbregistrering/enhet/adapter/OrganisasjonDetaljerDto.java
package no.nav.fo.veilarbregistrering.enhet.adapter;

import java.util.List;

import static java.util.Collections.emptyList;

public class OrganisasjonDetaljerDto {

    private List<ForretningsAdresseDto> forretningsadresser;
    private List<PostadresseDto> postadresser;

    OrganisasjonDetaljerDto() {
        //default constructor for Gson
    }

    public OrganisasjonDetaljerDto(List<ForretningsAdresseDto> forretningsadresser, List<PostadresseDto> postadresser) {
        this.forretningsadresser = forretningsadresser;
        this.postadresser = postadresser;
    }

    public void setForretningsadresser(List<ForretningsAdresseDto> forretningsadresser) {
        this.forretningsadresser = forretningsadresser;
    }

    public List<ForretningsAdresseDto> getForretningsadresser() {
        return forretningsadresser != null ? forretningsadresser : emptyList();
    }

    public void setPostadresser(List<PostadresseDto> postadresser) {
        this.postadresser = postadresser != null ? postadresser : emptyList();
    }

    public List<PostadresseDto> getPostadresser() {
        return postadresser != null ? postadresser : emptyList();
    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/enhet/adapter/OrganisasjonDetaljerDto.kt
package no.nav.fo.veilarbregistrering.enhet.adapter;

import java.util.List;

import static java.util.Collections.emptyList;

public class OrganisasjonDetaljerDto {

    private final List<ForretningsAdresseDto> forretningsadresser;
    private final List<PostadresseDto> postadresser;

    public OrganisasjonDetaljerDto(List<ForretningsAdresseDto> forretningsadresser, List<PostadresseDto> postadresser) {
        this.forretningsadresser = forretningsadresser;
        this.postadresser = postadresser;
    }

    public List<ForretningsAdresseDto> getForretningsadresser() {
        return forretningsadresser != null ? forretningsadresser : emptyList();
    }

    public List<PostadresseDto> getPostadresser() {
        return postadresser != null ? postadresser : emptyList();
    }
}